GCC error when script calls `make`

I changed the ENV VERSION in the Dockerfile.template file to 2.73.1+rev1.prod, then did a `git push balena master` to my Balena remote.

Unfortunately I get a GCC error ` gcc: error: unrecognized command line option '-mstack-protector-guard=sysreg'; did you mean '-fstack-protector-strong'?`

The hardware I'm using is a Raspberry Pi 4.

Any advice?

Here is the log:

```
main]     make: Entering directory '/tmp/tmp.U9THnSfxfP'
[main]       CC [M]  /usr/src/app/output/example_module_raspberrypi4-64_2.73.1+rev1.prod/hello.o
[main]     gcc: error: unrecognized command line option '-mstack-protector-guard=sysreg'; did you mean '-fstack-protector-strong'?
[main]     
[main]     gcc: error: unrecognized command line option '-mstack-protector-guard-reg=sp_el0'; did you mean '-fstack-protector-all'?
[main]     
[main]     gcc: error: unrecognized command line option '-mstack-protector-guard-offset=1336'; did you mean '-fstack-protector-strong'?
[main]     
[main]     make[1]: *** [scripts/Makefile.build:266: /usr/src/app/output/example_module_raspberrypi4-64_2.73.1+rev1.prod/hello.o] Error 1
[main]     
[main]     make: *** [Makefile:1732: /usr/src/app/output/example_module_raspberrypi4-64_2.73.1+rev1.prod] Error 2
[main]     
[main]     make: Leaving directory '/tmp/tmp.U9THnSfxfP'
[main]     Removing intermediate container e9788411e598
[main]     The command '/bin/sh -c BALENA_MACHINE_NAME=raspberrypi4-64 ./build.sh build --device raspberrypi4-64 --os-version "$VERSION" --src example_module' returned a non-zero code: 2
[Info]     Uploading images
[Success]  Successfully uploaded images
[Error]    Some services failed to build:
[Error]      Service: main
[Error]        Error: The command '/bin/sh -c BALENA_MACHINE_NAME=raspberrypi4-64 ./build.sh build --device raspberrypi4-64 --os-version "$VERSION" --src example_module' returned a non-zero code: 2
[Info]     Built on arm02
[Error]    Not deploying release.

```

--- a/Dockerfile.template
+++ b/Dockerfile.template
@@ -1,10 +1,14 @@
 FROM balenalib/%%BALENA_MACHINE_NAME%%-debian
 
-RUN apt-get update && apt-get install -y curl wget build-essential libelf-dev awscli bc flex libssl-dev python bison
+RUN install_packages curl wget build-essential libelf-dev awscli bc flex libssl-dev python bison
+
+# install gcc 10
+RUN echo 'deb http://deb.debian.org/debian testing main' >> /etc/apt/sources.list && install_packages gcc-9 libc6-dev
+
 COPY . /usr/src/app
 WORKDIR /usr/src/app
 
-ENV VERSION '2.47.0+rev1.prod 2.50.4+rev1.prod'
+ENV VERSION '2.73.1+rev1.prod'
 RUN BALENA_MACHINE_NAME=%%BALENA_MACHINE_NAME%% ./build.sh build --device %%BALENA_MACHINE_NAME%% --os-version "$VERSION" --src example_module
 
 CMD ./run.sh

--- a/Dockerfile.template
+++ b/Dockerfile.template
@@ -1,14 +1,23 @@
-FROM balenalib/%%BALENA_MACHINE_NAME%%-debian
+FROM balenalib/%%BALENA_MACHINE_NAME%%-debian-python:3-bullseye
 
-RUN install_packages curl wget build-essential libelf-dev awscli bc flex libssl-dev python bison
+RUN install_packages \
+    curl \
+    wget \
+    build-essential \
+    libelf-dev \
+    awscli \
+    bc \
+    flex \
+    libssl-dev \
+    bison
 
-# install gcc 10
-RUN echo 'deb http://deb.debian.org/debian testing main' >> /etc/apt/sources.list && install_packages awscli python3-yaml gcc-9 libc6-dev
 
 COPY . /usr/src/app
+
 WORKDIR /usr/src/app
 
-ENV VERSION '2.73.1+rev1.prod'
+ENV VERSION '2.88.4.prod'
+
 RUN BALENA_MACHINE_NAME=%%BALENA_MACHINE_NAME%% ./build.sh build --device %%BALENA_MACHINE_NAME%% --os-version "$VERSION" --src example_module
 
 CMD [ "/usr/src/app/run.sh" ]

--- a/Dockerfile.template
+++ b/Dockerfile.template
@@ -3,7 +3,7 @@ FROM balenalib/%%BALENA_MACHINE_NAME%%-debian
 RUN install_packages curl wget build-essential libelf-dev awscli bc flex libssl-dev python bison
 
 # install gcc 10
-RUN echo 'deb http://deb.debian.org/debian testing main' >> /etc/apt/sources.list && install_packages gcc-9 libc6-dev
+RUN echo 'deb http://deb.debian.org/debian testing main' >> /etc/apt/sources.list && install_packages awscli python3-yaml gcc-9 libc6-dev
 
 COPY . /usr/src/app
 WORKDIR /usr/src/app
@@ -11,4 +11,4 @@ WORKDIR /usr/src/app
 ENV VERSION '2.73.1+rev1.prod'
 RUN BALENA_MACHINE_NAME=%%BALENA_MACHINE_NAME%% ./build.sh build --device %%BALENA_MACHINE_NAME%% --os-version "$VERSION" --src example_module
 
-CMD ./run.sh
+CMD [ "/usr/src/app/run.sh" ]

